lib/luacrypto_acompress.c

```diff
@@ -0,0 +1,373 @@
+	return 0;										\
+}
+
+/***
+* Request object methods.
+* These methods are available on request objects created by `ACOMPRESS:request()`.
+* @type crypto_acomp_req
+*/
+
+/***
+* Compresses data asynchronously using the ACOMPRESS transform.
+* The callback is invoked when the operation completes (either synchronously or asynchronously).
+* @function compress
+* @tparam string data The data to compress.
+* @tparam integer output_size The maximum size of the output buffer.
+* @tparam function callback The callback function to invoke upon completion. It receives two arguments: `err` (integer error code, 0 on success) and `data` (string containing compressed data, or nil on error).
+* @raise Error if the request object is busy, or if parameters are invalid.
+*/
+LUACRYPTO_ACOMP_REQ_OPERATION(compress)
+
+/***
+* Decompresses data asynchronously using the ACOMPRESS transform.
+* The callback is invoked when the operation completes (either synchronously or asynchronously).
+* @function decompress
+* @tparam string data The compressed data to decompress.
+* @tparam integer output_size The maximum size of the output buffer.
+* @tparam function callback The callback function to invoke upon completion. It receives two arguments: `err` (integer error code, 0 on success) and `data` (string containing decompressed data, or nil on error).
+* @raise Error if the request object is busy, or if parameters are invalid.
+*/
+LUACRYPTO_ACOMP_REQ_OPERATION(decompress)
+
+static const luaL_Reg luacrypto_acomp_req_mt[] = {
+	{"compress", luacrypto_acomp_req_compress},
+	{"decompress", luacrypto_acomp_req_decompress},
+	{"__gc", lunatik_deleteobject},
+	{"__close", lunatik_closeobject},
+	{"__index", lunatik_monitorobject},
+	{NULL, NULL}
+};
+
+static const lunatik_class_t luacrypto_acomp_req_class = {
+	.name = "crypto_acomp_req",
+	.methods = luacrypto_acomp_req_mt,
+	.release = luacrypto_acomp_req_release,
+	.sleep = true,
+};
+
+/***
+* ACOMPRESS object methods.
+* These methods are available on ACOMPRESS TFM objects created by `crypto.acompress.new()`.
+* @see new
+* @type ACOMPRESS
+*/
+
+/***
+* Creates a new asynchronous compression request object.
+* Request objects are used to perform asynchronous compression and decompression operations.
+* Multiple requests can be created from the same TFM and used concurrently.
+* @function request
+* @treturn crypto_acomp_req A new request object.
+* @raise Error if request allocation fails.
+*/
+static int luacrypto_acompress_request(lua_State *L)
+{
+	struct crypto_acomp *tfm = luacrypto_acompress_check(L, 1);
+	lunatik_object_t *object;
+	luacrypto_acomp_req_t *req;
+
+	object = lunatik_newobject(L, &luacrypto_acomp_req_class, sizeof(luacrypto_acomp_req_t));
+	req = (luacrypto_acomp_req_t *)object->private;
+
+	memset(req, 0, sizeof(luacrypto_acomp_req_t));
+	req->runtime = lunatik_toruntime(L);
+	lunatik_getobject(req->runtime);
+	req->cb = LUA_NOREF;
+	req->buf = LUA_NOREF;
+	req->self = LUA_NOREF;
```

Comment:
I think using a lua table for all you need will simplify a bit the implementation..